adapters/repos/db/migrator.go

```diff
@@ -277,14 +292,39 @@ func (m *Migrator) updateIndexDeleteTenants(ctx context.Context,
 	return nil
 }
 
-func (m *Migrator) updateIndexAddShards(ctx context.Context, idx *Index,
+func (m *Migrator) updateIndexShards(ctx context.Context, idx *Index,
 	incomingSS *sharding.State,
 ) error {
-	for _, shardName := range incomingSS.AllLocalPhysicalShards() {
-		if err := idx.initLocalShard(ctx, shardName); err != nil {
-			return fmt.Errorf("add missing shard %s during update index: %w", shardName, err)
+	requestedShards := incomingSS.AllLocalPhysicalShards()
+	existedShards := make(map[string]ShardLike)
```

Comment:
I think some better names would help to make this code clearer, how about:
requestedShards => activeShards
existedShards => existingShards
?